PowerShell/JumpCloud Module/Public/Directory/Add-JCGsuiteMember.ps1

```diff
@@ -0,0 +1,106 @@
+function Add-JCGsuiteMember () {
+    param(
+        [Parameter( ValueFromPipelineByPropertyName, ParameterSetName = 'ByName', HelpMessage = 'The name of cloud directory instance')]
+        [String]$Name,
+        [Parameter( ValueFromPipelineByPropertyName, ParameterSetName = 'ByID', HelpMessage = 'The ID of cloud directory instance')]
+        [Alias('_id')]
+        [String]$ID,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A username to add to the directory')]
+        [String]$Username,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A UserID to add to the directory')]
+        [String]$UserID,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A UserGroup ID to add to the directory')]
+        [String]$GroupID,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A UserGroup name to add to the directory')]
+        [String]$GroupName
+    )
+    begin {
+        Write-Debug 'Verifying JCAPI Key'
+        if ($JCAPIKEY.length -ne 40) {
+            Connect-JCOnline
+        }
+        $resultsArray = [System.Collections.Generic.List[PSObject]]::new()
+        $DirectoryHash = Get-JcSdkDirectory | Where-Object type -EQ 'g_suite' | Select-Object id, name
+        if (($Username -or $UserID) -and ($GroupID -or $GroupName)) {
+            throw "Please add one type of association per call"
+        } elseif ($Username -and $UserID) {
+            throw "Please use either a username or a userID"
+        } elseif ($GroupID -and $GroupName) {
+            throw "Please use either a UserGroup Name or a UserGroup ID"
+        } elseif ($Username -or $UserID) {
+            Write-Debug 'Populating UserHash'
+            $UserHash = Get-DynamicHash -Object User -returnProperties username
+        } elseif ($GroupID -or $GroupName) {
+            Write-Debug 'Populating UserGroupHash'
+            $UserGroupHash = Get-DynamicHash -Object Group -GroupType User -returnProperties name
+        }
+    }
+    process {
+        switch ($PSCmdlet.ParameterSetName) {
+            ByName {
+                Write-Debug "Finding directory by Name"
+                $CloudDirectory = $DirectoryHash | Where-Object name -EQ $Name
+                if (!$CloudDirectory) {
+                    throw "$Name was not found. Please try again"
+                } elseif ($CloudDirectory.count -gt 1) {
+                    throw "Multiple directories with the same name detected, please use the -id parameter to specify which directory to edit"
```

Comment:
Thanks for this, it's something we would have had to address, I doubt many folks will have this problem but good to catch it before it comes up